app/controllers/course_summaries_controller.rb

```diff
@@ -46,6 +46,33 @@ def populate
     }
   end
 
+  def grade_distribution
+    table_data = []
+    marking_schemes = current_user.student? ? MarkingScheme.none : MarkingScheme
+
+    marking_schemes.order(id: :asc).each do |m|
```

Comment:
Use map instead of each, since the former returns an array, so you don't need to `table_data <<` manually in the block.